src/main/java/com/faforever/api/user/GogService.java

```diff
@@ -0,0 +1,177 @@
+    String profilePageHtml = null;
+
+    try {
+      profilePageHtml = new RestTemplate().getForObject(profilePageUrl, String.class);
+    } catch (HttpClientErrorException e) {
+      if (e.getRawStatusCode() == HttpStatus.NOT_FOUND.value()) {
+        throw ApiException.of(ErrorCode.GOG_LINK_PROFILE_NOT_PUBLIC);
+      } else {
+        log.error("Couldn't retrieve gog profile page for {}", profilePageUrl);
+        throw ApiException.of(ErrorCode.GOG_LINK_INTERNAL_SERVER_ERROR);
+      }
+    }
+
+    if (profilePageHtml == null) {
+      log.error("Couldn't retrieve gog profile page for {}", profilePageUrl);
+      throw ApiException.of(ErrorCode.GOG_LINK_INTERNAL_SERVER_ERROR);
+    }
+
+    Document document = Jsoup.parse(profilePageHtml);
+    for (Element element : document.body().getElementsByTag("script")) {
+      String scriptText = element.data();
+      Matcher matcher = PROFILE_USER_STATUS_PATTERN.matcher(scriptText);
+
+      if (matcher.find()) {
+        return matcher.group(1);
+      }
+    }
+
+    throw ApiException.of(ErrorCode.GOG_LINK_PROFILE_NOT_PUBLIC);
+  }
+
+  boolean verifyGameOwnership(String gogUsername) {
+    List<GogGamesListPage.GogGamesListEntry> allGames = getAllGames(gogUsername);
+
+    return allGames.stream().anyMatch(game -> Objects.equals(game.game.id, GOG_FA_GAME_ID));
+  }
+
+  private List<GogGamesListPage.GogGamesListEntry> getAllGames(String gogUsername) {
+    GogGamesListPage firstPage = getGamesListPage(gogUsername, 1);
+
+    List<GogGamesListPage> pages = new ArrayList<>();
+    pages.add(firstPage);
+    for (int i = 2; i <= firstPage.getPages(); i++) {
+      pages.add(getGamesListPage(gogUsername, i));
+    }
+
+    return pages.stream().flatMap(page -> page.get_embedded().getItems().stream()).collect(Collectors.toList());
+  }
+
+  private GogGamesListPage getGamesListPage(String gogUsername, int page) {
+    String gamesListUrl = String.format(properties.getGog().getGamesListUrl(), gogUsername, page);
+
+    try {
+      String json = new RestTemplate().getForObject(gamesListUrl, String.class);
+      if (json == null || json.isBlank()) {
+        throw ApiException.of(ErrorCode.GOG_LINK_GAMES_NOT_PUBLIC);
+      }
+      return gson.fromJson(json, GogGamesListPage.class);
+    } catch (HttpClientErrorException e) {
+      if (e.getRawStatusCode() == HttpStatus.FORBIDDEN.value()) {
+        throw ApiException.of(ErrorCode.GOG_LINK_GAMES_NOT_PUBLIC);
+      } else if (e.getRawStatusCode() == HttpStatus.NOT_FOUND.value()) {
+        throw ApiException.of(ErrorCode.GOG_LINK_USER_NOT_FOUND);
+      } else {
+        throw e;
+      }
+    }
+  }
+
+  @Data
+  @AllArgsConstructor
+  private static class GogGamesListPage implements Serializable {
+    private int page;
+    private int limit;
+    private int pages;
+    private int total;
+    private GogGamesListLinkList _links;
+    private GogGamesListEmbeddedList _embedded;
```

Comment:
use proper java variable names and use @JsonProperty to indicate the json field name